### ISSUE TITLE
multi-stage builds fail when SSL certificates are mounted on read-only filesystem

### ISSUE THREAD

**Issue by diclophis** [GoogleContainerTools/kaniko#222]
Hello,

I currently am getting this error when attempting to use `kaniko` to build a multi-stage `Dockerfile`

```
# normal stuff ...
time="2018-07-03T06:04:05Z" level=info msg="Taking snapshot of full filesystem..."
time="2018-07-03T06:04:06Z" level=info msg="cmd: copy [CHANGELOG.md]"
time="2018-07-03T06:04:06Z" level=info msg="dest: /CHANGELOG.md"
time="2018-07-03T06:04:06Z" level=info msg="Copying file /var/tmp/git/repo/CHANGELOG.md to /CHANGELOG.md"
time="2018-07-03T06:04:06Z" level=info msg="cmd: /bin/sh"
time="2018-07-03T06:04:06Z" level=info msg="args: [-c true]"
2018/07/03 06:04:06 No matching credentials found for index.docker.io, falling back on anonymous
time="2018-07-03T06:04:07Z" level=info msg="saving dependencies [/CHANGELOG.md]"
time="2018-07-03T06:04:07Z" level=info msg="Copying file /CHANGELOG.md to /kaniko/0/CHANGELOG.md"
time="2018-07-03T06:04:07Z" level=info msg="Deleting filesystem..."
time="2018-07-03T06:04:07Z" level=error msg="remove /kaniko/ssl/certs/ca-certificates.crt: read-only file system"
pod default/kaniko-test-wkndr terminated (Error)
```

I have deployed `kaniko` using approx. these manifests:

```
---
apiVersion: v1
metadata:
  annotations:
    seccomp.security.alpha.kubernetes.io/pod: 'docker/default'
spec:
  initContainers:
    - name: git-clone
      image: something-with-git-and-bash:latest
      imagePullPolicy: IfNotPresent
      args:
        - bash
        - -c
        - git clone http://dumb:8080/repo /var/tmp/git/repo && cd /var/tmp/git/repo && git checkout master
      securityContext:
        runAsUser: 1
        allowPrivilegeEscalation: false
        readOnlyRootFilesystem: true
      volumeMounts:
        - mountPath: /var/tmp/git
          name: git-repo
  containers:
  - image: gcr.io/kaniko-project/executor:latest
    imagePullPolicy: IfNotPresent
    name: kaniko-build
    args: [
            "--dockerfile", "Dockerfile",
            "--destination", "registry:5000/repo:kaniko-latest",
            "-c", "/var/tmp/git/repo"
          ]
    volumeMounts:
    - mountPath: /var/tmp/git
      name: git-repo
    - mountPath: /kaniko/ssl/certs
      name: ca-certificates
  volumes:
  - name: git-repo
    emptyDir: {}
  - configMap:
      name: ca-certificates
    name: ca-certificates
...
```

Using this `Dockerfile`

```
FROM ubuntu:bionic-20180526
COPY CHANGELOG.md /CHANGELOG.md
RUN true

FROM alpine:3.7
COPY --from=0 /CHANGELOG.md /CHANGELOG-SECOND.md
RUN true
```

And of course a repo with the `Dockerfile` and a `CHANGELOG.md` file...

Other relevant information?

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.0", GitCommit:"925c127ec6b946659ad0fd596fa959be43f0cc05", GitTreeState:"clean", BuildDate:"2017-12-15T21:07:38Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:05:37Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by priyawadhwa** [comment]
Hey @diclophis, I can't repro this issue. This is the pod spec I'm running, but it's able to build the same image successfully:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: kaniko
  annotations:
    seccomp.security.alpha.kubernetes.io/pod: 'docker/default'
spec:
  containers:
  - image: gcr.io/priya-wadhwa/executor
    imagePullPolicy: IfNotPresent
    name: kaniko-build
    args: [
            "--dockerfile", "Dockerfile",
            "--destination", "gcr.io/priya-wadhwa/test:test",
            "-c", "gs://kanikotest"
          ]
    volumeMounts:
    - name: ca-certificates 
      mountPath: /kaniko/ssl/certs
    - name: kaniko-secret
      mountPath: /secret
    env:
      - name: GOOGLE_APPLICATION_CREDENTIALS
        value: /secret/kaniko-secret.json
  volumes:
  - name: kaniko-secret
    secret:
      secretName: kaniko-secret
  - configMap:
      name: ca-certificates
    name: ca-certificates
```

I did find a different bug when trying to figure this out though, which was fixed here (#228). I released a new version of kaniko, could you see if you still get the same error using that?

**Comment by diclophis** [comment]
yea, it appears to be working with the latest build `5f9fb2cb8d55003b2da8f406e22a4705b08d471e` thanks!
